python/dgl/graphbolt/subgraph_sampler.py

```diff
@@ -0,0 +1,31 @@
+"""Subgraph samplers"""
+
+from torchdata.datapipes.iter import IterDataPipe
+
+
+class SubgraphSampler(IterDataPipe):
+    """A subgraph sampler.
+
+    It is an iterator equivalent to the following:
+
+    .. code:: python
+
+       for data in dp:
+           yield sampler_func(data)
+
+    Parameters
+    ----------
+    dp : DataPipe
+        The datapipe.
+    sampler_func : callable
+        The subgraph sampling function.
+    """
+
+    def __init__(self, dp, sampler_func):
```

Comment:
```suggestion
    def __init__(self, data_pipe, sampler_func):
```

Reply:
No longer need any code for initialization.